### PR TITLE
Print stderr in case clone fails and add toolbox functions to retry ommands

### PR DESCRIPTION
**Pull request by hamir-suspect**
- in case checkout fails to clone prints out error message 
- checkout can be called with `retry checkout`

### CODE REVIEW

File: retry [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #   -t, --times  - Number of times to retry the command (default: 3)
 #   -s, --sleep  - Number of seconds to sleep between retries (default: 0)
 #
+source ~/.toolbox/toolbox
```
Comment (lucaspin) [inline retry:11]: Why is this needed?

File: libcheckout [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#!/bin/bash
```
Comment (lucaspin) [inline libcheckout:1]: Why is this needed?

File: retry (hunk 1 above)
Comment (hamir-suspect) [reply retry:11]: For retry to work with other functions that are scoped, like `checkout`, it needs to source them. 
eg. if you try naive `retry checkout` it will not work since checkout is using namespaced functions like `checkout::shallow()` which is not available.

**Review by DamjanBecirovic — APPROVED** [review]
